pkg/windows/windows.go

```diff
@@ -417,50 +419,27 @@ func (vm *windows) EnsureRequiredServicesStopped() error {
 	return nil
 }
 
-// ensureServicesAreRemoved ensures that all services installed by WMCO are removed from the instance
-func (vm *windows) ensureServicesAreRemoved() error {
-	// TODO: In this transitional period of migrating services to WICD's responsibility, WICD must be stopped first to
-	//       ensure it does not start any of these stopped services. This code should be removed as part of
-	//       https://issues.redhat.com/browse/WINC-733
-	svc := &service{name: wicdServiceName}
-	if err := vm.ensureServiceNotRunning(svc); err != nil {
-		return errors.Wrapf(err, "could not stop service %s", wicdServiceName)
-	}
-	for _, svcName := range append(RequiredServices, AzureCloudNodeManagerServiceName) {
-		svc := &service{name: svcName}
-
-		// If the service is not installed, do nothing
-		exists, err := vm.serviceExists(svc.name)
-		if err != nil {
-			return errors.Wrapf(err, "unable to check if %s service exists", svc.name)
-		}
-		if !exists {
-			continue
-		}
+func (vm *windows) Deconfigure(preserveNode bool, apiServerURL string, credentials *Authentication) error {
+	vm.log.Info("deconfiguring")
 
-		// Make sure the service is stopped before we attempt to delete it
-		if err := vm.ensureServiceNotRunning(svc); err != nil {
-			return errors.Wrapf(err, "could not stop service %s", svc.name)
-		}
-		if err := vm.deleteService(svc); err != nil {
-			return errors.Wrapf(err, "could not delete service %s", svcName)
-		}
-		vm.log.Info("deconfigured", "service", svc.name)
+	if err := vm.ensureWICDSecretContent(credentials); err != nil {
+		return err
 	}
-	return nil
-}
-
-func (vm *windows) Deconfigure() error {
-	vm.log.Info("deconfiguring")
-	if err := vm.ensureServicesAreRemoved(); err != nil {
-		return errors.Wrap(err, "unable to remove Windows services")
+	if err := vm.ensureHNSNetworksAreRemoved(); err != nil {
+		return errors.Wrap(err, "unable to ensure HNS networks are removed")
+	}
+	wicdCleanupCmd := fmt.Sprintf("%s cleanup --api-server %s --sa-ca %s --sa-token %s",
```

Comment:
There should be a `DeconfigureWICD()` call equivalent to the `ConfigureWICD()` call before calling clean up. WICD shouldn't be cleaning up itself. In fact the `cleanup` and `bootstrap` commands should fail if the WICD service is running.

Reply:
That's a good point @saifshaikh48